_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder_va.cpp

```diff
@@ -0,0 +1,186 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "umc_defs.h"
+#ifdef MFX_ENABLE_AV1_VIDEO_DECODE
+
+#include "umc_structures.h"
+#include "umc_va_base.h"
+
+#include "umc_av1_decoder_va.h"
+#include "umc_av1_utils.h"
+#include "umc_av1_frame.h"
+#include "umc_av1_bitstream.h"
+#include "umc_av1_va_packer.h"
+
+#include "umc_frame_data.h"
+
+#include <algorithm>
+
+namespace UMC_AV1_DECODER
+{
+    AV1DecoderVA::AV1DecoderVA()
+        : va(nullptr)
+    {}
+
+    UMC::Status AV1DecoderVA::SetParams(UMC::BaseCodecParams* info)
+    {
+        if (!info)
+            return UMC::UMC_ERR_NULL_PTR;
+
+        AV1DecoderParams* dp =
+            DynamicCast<AV1DecoderParams, UMC::BaseCodecParams>(info);
+        if (!dp)
+            return UMC::UMC_ERR_INVALID_PARAMS;
+
+        if (!dp->pVideoAccelerator)
+            return UMC::UMC_ERR_NULL_PTR;
+
+        va = dp->pVideoAccelerator;
+        packer.reset(Packer::CreatePacker(va));
+
+        uint32_t const dpb_size =
+            params.async_depth + TOTAL_REFS;
+        SetDPBSize(dpb_size);
+        SetRefSize(TOTAL_REFS);
+        return UMC::UMC_OK;
+    }
+
+    UMC::Status AV1DecoderVA::SubmitTiles(AV1DecoderFrame& frame, bool firstSubmission)
+    {
+        VM_ASSERT(va);
+        UMC::Status sts = UMC::UMC_OK;
+
+        if (firstSubmission)
+        {
+            // it's first submission for current frame - need to call BeginFrame
+            sts = va->BeginFrame(frame.GetMemID()); // TODO: [Rev0.85] Clarify with Ce which surface index to send: recon of display
+            if (sts != UMC::UMC_OK)
+                return sts;
+
+            VM_ASSERT(packer);
+            packer->BeginFrame();
+
+            frame.StartDecoding();
+        }
+
+        auto &tileSets = frame.GetTileSets();
+        packer->PackAU(tileSets, frame, firstSubmission);
+
+        const bool lastSubmission = (GetNumMissingTiles(frame) == 0);
+        if (lastSubmission)
+            packer->EndFrame();
+
+        sts = va->Execute();
+
+        if (lastSubmission) // it's last submission for current frame - need to call EndFrame
+            sts = va->EndFrame();
+
+        return sts;
+    }
+
+    void AV1DecoderVA::AllocateFrameData(UMC::VideoDataInfo const& info, UMC::FrameMemID id, AV1DecoderFrame& frame)
+    {
+        VM_ASSERT(id != UMC::FRAME_MID_INVALID);
+
+        UMC::FrameData fd;
+        fd.Init(&info, id, allocator);
+
+        frame.AllocateAndLock(&fd);
+    }
+
+    inline bool InProgress(AV1DecoderFrame const& frame)
+    {
+        return frame.DecodingStarted() && !frame.DecodingCompleted();
+    }
+
+    inline void SetError(AV1DecoderFrame& frame, uint8_t status)
+    {
+        switch (status)
+        {
+        case 1:
+            frame.AddError(UMC::ERROR_FRAME_MINOR);
+            break;
+        case 2:
+        case 3:
+        case 4:
```

Comment:
Please avoid constants . Give them names

Reply:
I double checked it. This function seems not be called. I will remove it.